tests/models/test_openai.py

```diff
@@ -822,6 +823,104 @@ async def test_document_url_input(allow_model_requests: None, openai_api_key: st
     assert result.output == snapshot('The document contains the text "Dummy PDF file" on its single page.')
 
 
+@pytest.mark.parametrize(
+    'media_type,body',
+    [
+        ('text/plain', 'Hello'),
+        ('application/json', '{"a":1}'),
+        ('application/xml', '<a>1</a>'),
+        ('application/yaml', 'a: 1'),
+    ],
+)
+async def test_openai_binary_content_text_like_is_inlined(
+    media_type: str, body: str, openai_api_key: str, allow_model_requests: None
+) -> None:
+    # Arrange input
+    bin_content = BinaryContent(data=body.encode(), media_type=media_type)
+    identifier = bin_content.identifier
+
+    # Capture mapped OpenAI messages via public request() API
+    captured: list[list[dict[str, Any]]] = []
+
+    async def fake_create(*args: Any, **kwargs: Any):
+        captured.append(kwargs['messages'])
+        raise RuntimeError('stop-after-capture')
+
+    model = OpenAIChatModel('gpt-4o', provider=OpenAIProvider(api_key=openai_api_key))
+    # Monkeypatch the client's create method
+    model.client.chat.completions.create = fake_create
```

Comment:
In this particular case I think it's acceptable to directly call the private `_map_user_message` method (and add a `# type: ignore[reportPrivateUsage]` comment) as it's a lot easier to follow than this mocking.

Reply:
Done